static/app/utils/discover/fields.tsx

```diff
@@ -23,16 +25,9 @@ export type Field = {
   width?: number;
 };
 
-export type ColumnType =
-  | 'boolean'
-  | 'date'
-  | 'duration'
-  | 'integer'
-  | 'number'
-  | 'percentage'
-  | 'string';
+export type ColumnType = `${Exclude<FieldValueType, FieldValueType.NEVER>}`;
```

Comment:
Does it actually need to be in the string template for the type to be coerced into a string?

Reply:
Annoyingly yes, if I don't:

![Screen Shot 2022-07-20 at 12 44 52 PM](https://user-images.githubusercontent.com/30991498/180068568-c6b647dc-e4eb-4b13-b435-e9190cb1516b.png)
![Screen Shot 2022-07-20 at 12 45 16 PM](https://user-images.githubusercontent.com/30991498/180068540-4e326e65-98da-49f0-a539-9f63a46654c6.png)

Reply:
The error will go away if you use `[FieldValueType.STRING, FieldValueType.DURATION, FieldValueType.NUMBER]`, is there a reason we can't do that?

Reply:
Oh, I see. There are hundreds of existing uses with string literals.

I myself would prefer those to use the enum (makes refactoring easier), but if we don't want to make all those changes a comment might suffice here explaining why we need ColumnType to be a string literal @jennmueng

Reply:
@malwilley Definitely agree and would really prefer to use the enum and refactor everything, just that I'm already running low on time before my internship is done haha